### ISSUE TITLE
[Edit] Error when creating in Edit mode

### ISSUE THREAD

**Issue by VWoeltjen** [nasa/openmctweb#305]
Steps to reproduce:
* Create a Timeline
* Go to Edit mode
* Create an Activity using the Add button from the toolbar

Observe on console:

```
TypeError: Cannot read property 'then' of undefined
    at :8080/platform/commonUI/browse/src/creation/CreationService.js:89
    at :8080/platform/framework/lib/angular.min.js:118
    at n.$eval (:8080/platform/framework/lib/angular.min.js:133)
    at n.$digest (:8080/platform/framework/lib/angular.min.js:130)
    at :8080/platform/framework/lib/angular.min.js:133
    at e (:8080/platform/framework/lib/angular.min.js:42)
    at :8080/platform/framework/lib/angular.min.js:45
```

Root cause:
* EditablePersistenceCapability does not return a promise from `persist`, which is inconsistent with the persistence capability's interface.

Notably, no incorrect behavior is immediately visible within the client as a consequence of this, but there may be other situations where this is not the case.

**Comment by VWoeltjen** [comment]
Partial fix on open305. Resolving this causes another problem to be tickled: The newly-created object references its parent as its location, _but_ the actual parent does not contain it within its composition. When this gets contextualized (by the LocatingObjectDecorator) a warning is issued:

```
Attempted to contextualize 18998ddb-440a-48c1-983b-0cbbb02ddab6 in 9ad4baa4-7764-4e66-8a48-84a08cd4dae3 but that object does not contain 18998ddb-440a-48c1-983b-0cbbb02ddab6 in its composition. Unexpected behavior may follow.
```

This is actually benign in this case. Some options:

1. Check for this case from LocatingObjectDecorator, and just don't contextualize objects in this case. This should still behave correctly under nominal circumstances, but is funny logic.
2. Modify Edit mode such that objects created during Edit mode are not actually persisted until Edit mode completes. This means that "temporary" objects in Edit mode need to be retrievable somehow; can probably stuff this into the EditableLookupCapability.
3. Just live with the grumpy logging for now.

Option 1 sounds funny. Option 2 sounds most correct, but don't know how much effort to put into this with NEM on the horizon.

Think Option 2 should be pursued, falling back to Option 3 if it is non-trivial.

**Comment by VWoeltjen** [comment]
Seeing very weird behavior when methods are wrapped for editability; in particular, [this line](https://github.com/nasa/openmctweb/blob/master/platform/commonUI/edit/src/capabilities/EditableLookupCapability.js#L117):

`Object.keys(contextCapability).forEach(wrapMethod);`

...makes no sense since recent code style changes. The intent here is to wrap all methods exposed by the relevant object, such that any domain objects that get returned (either directly or via promises) get wrapped as editable domain objects.

What happens instead is that any private members that happen to be functions get wrapped. This happens to work for `composition` because it maintains a reference to the `contextualize` service in this fashion (so any objects contextualized via `composition` also end up becoming editable.) It appears that `context` is not wrapped appropriately; as a consequence, changes in the Library (e.g. via Edit Properties) get persisted immediately.

Think this can be reasonably quickly resolved by specifying which methods to wrap for which capabilities. This is less maintainable but with new Edit Mode on the way, unclear how long this code will need to be maintained anyway.

**Comment by RCarvalhoNASA** [comment]
Testathon 20151207 - These errors are invisible to the normal user, but there are still errors showing up in the console when one opens that on the browser.

**Comment by charlesh88** [comment]
I also see console errors, see screenshot. All lines below the cursor arrow were new when adding an Activity, including the first error.
![screenshot-developer tools - http --10 10 81 15 9080-static-index html](https://cloud.githubusercontent.com/assets/1056412/11643973/a3774768-9cfc-11e5-8e0c-1f8c4acc0b9f.png)

**Comment by VWoeltjen** [comment]
Moving out of Baxter; remaining messaging is benign and won't be seen by the user. Issue may or may not be made obsolete by new Edit Mode

**Comment by larkin** [comment]
@VWoeltjen Object.keys(contextCapability).forEach(wrapMethod);
It's not wrapping private methods.  It's wrapping `own` methods.  Prototype methods are not enumerated.

From MDN:
> The `Object.keys()` method returns an array of a given object's own enumerable properties, in the same order as that provided by a `for...in` loop (the difference being that a for-in loop enumerates properties in the prototype chain as well).

**Comment by VWoeltjen** [comment]
Reassigning to @akhenry to see if this can/should be fixed in the context of New Edit Mode

**Comment by akhenry** [comment]
> Seeing very weird behavior when methods are wrapped for editability; in particular, this line:
> `Object.keys(contextCapability).forEach(wrapMethod);`

Have resolved this in #500 

Warning messages regarding contextualization will need to be dealt with separately. I have created issue #575 to investigate this.

**Comment by akhenry** [comment]
Going to close this issue as it got a bit off topic and the originally reported error was resolved. Have created a new issue for the outstanding contextualization warnings.

https://github.com/nasa/openmctweb/issues/624
